pkgs/top-level/all-packages.nix

```diff
@@ -2574,6 +2574,7 @@ with pkgs;
   cudatoolkit = cudaPackages.cudatoolkit;
 
   curlFull = curl.override {
+    c-aresSupport = true;
```

Comment:
`c-ares` is very light, so I think we can have this for the regular `curl` derivation, instead of just `curlFull`